File: algo-credit-check/js/creditCheck.js
function sumStringArr(arr) {
    let sum = 0;
    for (let elem of arr) {
        sum += Number(elem)
    }
    return sum;
}

function sumNumArr(arr) {
    let sum = 0;
    for (let elem of arr) {
        sum += elem
    }
    return sum;
}

function creditCheck(str) {
    const numArr = str.split("");
    for (let i = 0; i < numArr.length; i++) {
        numArr[i] = Number(numArr[i]);
    }
    for (let i = 0; i < numArr.length; i += 2) {
        numArr[i] *= 2;
        if (numArr[i] > 9) {
            const newArr = numArr[i].toString().split("");
            numArr[i] = sumStringArr(newArr);
        }
    }
    const total = sumNumArr(numArr);
    if (total % 10 === 0) {
        return "The number is valid!";
    } else {
        return "The number is invalid!";
    }
}

module.exports = creditCheck;

File: algo-credit-check/python/credit_check.py
def sum_string_list(list):
    sum = 0
    for elem in list:
        sum += int(elem)
    return sum

def sum_number_list(list):
    sum = 0
    for elem in list:
        sum += elem
    return sum

def credit_check(string):
    num_list = [*string]
    for index, string in enumerate(num_list):
        num_list[index] = int(string)
    counter = 0
    while counter < len(num_list):
        num_list[counter] *= 2
        if num_list[counter] > 9:
            new_list = [*str(num_list[counter])]
            num_list[counter] = sum_string_list(new_list)
        counter += 2
    print(num_list)
    total = sum_number_list(num_list)
    if total % 10 == 0:
        return "The number is valid!"
    else:
        return "The number is invalid!"